src/connector/src/source/iceberg/parquet_file_handler.rs

```diff
@@ -265,7 +275,13 @@ pub fn extract_valid_column_indices(
             })
     })
     .collect();
-    Ok(valid_column_indices)
+            Ok(ProjectionMask::leaves(
```

Comment:
ProjectionMask has `ProjectionMask::roots` method. Is it what you want? In our forked iceberg-rust, we use this method to handle nested types. https://github.com/risingwavelabs/iceberg-rust/blob/683fb89edeaf8d1baae69e1f376d68b92be1d496/crates/iceberg/src/arrow/reader.rs#L387

Reply:
Yes, this is what I wanted. Thanks for letting me know. I have modified it.